plugin/trino-hive/src/main/java/io/trino/plugin/hive/aws/athena/projection/EnumProjection.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.aws.athena.projection;
+
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.spi.predicate.Domain.singleValue;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class EnumProjection
+        extends io.trino.plugin.hive.aws.athena.projection.Projection
+{
+    private List<String> values;
+
+    public EnumProjection(String columnName, List<String> values)
+    {
+        super(columnName);
+        this.values = requireNonNull(values, "values is null");
+    }
+
+    @Override
+    public List<String> getProjectedValues(Optional<Domain> partitionValueFilter)
+    {
+        if (partitionValueFilter.isEmpty() || partitionValueFilter.get().isAll()) {
+            return values;
+        }
+        return values.stream()
+                .filter(value -> isValueInDomain(partitionValueFilter, value))
+                .collect(toList());
+    }
+
+    private boolean isValueInDomain(Optional<Domain> valueDomain, String value)
+    {
+        Domain domain = valueDomain.get();
+        Type type = domain.getType();
+        if (type instanceof VarcharType) {
+            return domain.contains(singleValue(type, utf8Slice(value)));
+        }
+        return false;
```

Comment:
No in such case it means there is no need to project other partitions as there is an explicit predicate in query to a limited subset of values from our defined enum.

As for types:
- Char 🤔 I'm wondering. In Athena docs isn't explicitly saying that Enum type support strings only. I just follow fact that Enums.name() is always String.  
- Varbinary - hmm hard to tell to me. But sounds like a rare case when partitioning would be build basing on arbitrary length field stored as binary.

In general if it will ever show up that we need other data types support ti would be easy to extend and add.
But let's see if users will require it.